common/djangoapps/third_party_auth/samlproviderconfig/tests/test_samlproviderconfig.py

```diff
@@ -67,6 +65,12 @@ def test_get_one_config_by_enterprise_uuid_found(self):
         """
         GET auth/saml/v0/provider_config/?enterprise_customer_uuid=id=id
         """
+
+        # for GET to work, we need an association present
+        EnterpriseCustomerIdentityProvider.objects.get_or_create(
```

Comment:
Good question. I moved it because I realized that only this one endpoint actually needs this setup and actually it may cause 'noise' in other tests if I have this setup. However, if in future most tests ends up need this we can move it back up to setup

Reply:
They're both under the same class method though so how would setupTestData be called by others? The reasoning makes sense, I'm just unclear how setUp vs setUpTestData under the same `SAMLProviderConfigTests` class get called differently